contrib/src/main/java/com/datatorrent/contrib/avro/AvroFileInputOperator.java

```
@@ -0,0 +1,159 @@
+
+import com.datatorrent.api.AutoMetric;
+import com.datatorrent.api.DefaultOutputPort;
+import com.datatorrent.api.annotation.OutputPortFieldAnnotation;
+import com.datatorrent.lib.io.fs.AbstractFileInputOperator;
+
+/**
+ * <p>
+ * Avro File Input Operator
+ * </p>
+ * A specific implementation of the AbstractFileInputOperator to read Avro
+ * container files.<br>
+ * No need to provide schema,its inferred from the file<br>
+ * Users can add the {@link IdempotentStorageManager.FSIdempotentStorageManager}
+ * to ensure exactly once semantics with a HDFS backed WAL.
+ * 
+ * @displayName AvroFileInputOperator
+ * @category Input
+ * @tags fs, file,avro, input operator
+ * @since 3.3.0
+ */
+@InterfaceStability.Evolving
+public class AvroFileInputOperator extends AbstractFileInputOperator<GenericRecord>
+{
+
+  private transient long offset = 0L;
+
+  @AutoMetric
+  int recordCnt = 0;
+
+  @AutoMetric
+  int errorCnt = 0;
+
+  private transient DataFileStream<GenericRecord> avroDataStream;
+  private transient GenericRecord record = null;
+
+  public final transient DefaultOutputPort<GenericRecord> output = new DefaultOutputPort<GenericRecord>();
+
+  @OutputPortFieldAnnotation(optional = true)
+  public final transient DefaultOutputPort<String> completedFilesPort = new DefaultOutputPort<String>();
+
+  @OutputPortFieldAnnotation(optional = true)
+  public final transient DefaultOutputPort<String> errorRecordsPort = new DefaultOutputPort<String>();
+
+  @Override
+  protected InputStream openFile(Path path) throws IOException
+  {
+    InputStream is = super.openFile(path);
+    if (is != null) {
+      try {
+        DatumReader<GenericRecord> datumReader = new GenericDatumReader<GenericRecord>();
+        avroDataStream = new DataFileStream<GenericRecord>(is, datumReader);
+        datumReader.setSchema(avroDataStream.getSchema());
+        return is;
+      } catch (NullPointerException npe) {
+        LOG.error("Schemaless file - " + npe.getMessage());
```

Please don't hide any exception with logs... You could log the exception as well which would show callstack for debugging purpose...